src/IO/Resources/ArtLoader.cs

```diff
@@ -139,102 +153,126 @@ private void AddSpriteToAtlas(TextureAtlas atlas, int g, bool isTerrain)
+                        if ((fixedGraphic >= 0x2053 && fixedGraphic <= 0x2062) || (fixedGraphic >= 0x206A && fixedGraphic <= 0x2079))
                         {
-                            // keep the cursor graphic check to cleanup edges
-                            if ((fixedGraphic >= 0x2053 && fixedGraphic <= 0x2062) || (fixedGraphic >= 0x206A && fixedGraphic <= 0x2079))
+                            for (int i = 0; i < width; i++)
                             {
-                                for (int i = 0; i < width; i++)
-                                {
-                                    artPixels[i] = 0;
-                                    artPixels[(height - 1) * width + i] = 0;
-                                }
-
-                                for (int i = 0; i < height; i++)
-                                {
-                                    artPixels[i * width] = 0;
-                                    artPixels[i * width + width - 1] = 0;
-                                }
+                                data[i] = 0;
+                                data[(height - 1) * width + i] = 0;
                             }
 
-                            ref var spriteInfo = ref _spriteInfos[g];
+                            for (int i = 0; i < height; i++)
+                            {
+                                data[i * width] = 0;
+                                data[i * width + width - 1] = 0;
+                            }
+                        }
 
-                            FinalizeData
-                            (
-                                artPixels,
-                                ref entry,
-                                fixedGraphic,
-                                width,
-                                height,
-                                out spriteInfo.ArtBounds
-                            );
+                        ref var spriteInfo = ref _spriteInfos[g];
 
-                            _picker.Set(fixedGraphic, width, height, artPixels);
-                            spriteInfo.Texture = atlas.AddSprite(artPixels, width, height, out spriteInfo.UV);
-                        }
-                    }
-                    finally
-                    {
-                        if (buffer != null)
-                        {
-                            System.Buffers.ArrayPool<uint>.Shared.Return(buffer, true);
-                        }
+                        FinalizeData
+                        (
+                            data,
+                            ref entry,
+                            fixedGraphic,
+                            width,
+                            height,
+                            out spriteInfo.ArtBounds
+                        );
                     }
                 }
             }
+
+            return true;
         }
-      
+
+        [ThreadStatic] private static uint[] _data = new uint[0x10000];
```

Comment:
`[ThreadStatic]` causes to create 2 instances of `_decompressedData`:
1 at loading time due to the `Task.Run`
1 from the main thread

Reply:
Yes - that's by design so that both threads can load sprites. I can make it start off as null though so if a thread never attempts to read anything this doesn't get allocated.